lib/src/screens/activity/widgets/activity_card.dart

```diff
@@ -30,39 +29,28 @@ class ActivityCard extends StatelessWidget {
     String subtitleTranslationKey = '';
     String? logo;
 
-    switch (logEntry.type) {
-      case LogEntryType.disclosing:
-      case LogEntryType.signing:
-        if (logEntry.serverName != null) {
-          title = logEntry.serverName!.name.translate(lang);
-          if (logEntry.serverName!.logo != null) {
-            logo = logEntry.serverName!.logo;
-          }
-        }
-        subtitleTranslationKey =
-            logEntry.type == LogEntryType.disclosing ? 'activity.data_shared' : 'activity.message_signed';
-        break;
-      case LogEntryType.issuing:
-        final cred = Credential.fromRaw(
-          irmaConfiguration: irmaConfiguration,
-          rawCredential: logEntry.issuedCredentials.first,
-        );
-        if (irmaConfiguration.issuers[cred.info.issuer.fullId] != null) {
-          title = irmaConfiguration.issuers[cred.info.issuer.fullId]!.name.translate(lang);
+    if (logEntry.type == LogEntryType.removal) {
+      final credType = irmaConfiguration.credentialTypes[logEntry.removedCredentials.keys.first]!;
+      title = irmaConfiguration.issuers[credType.fullIssuerId]!.name.translate(lang);
+      subtitleTranslationKey = 'activity.data_deleted';
+      if (credType.logo != null) {
+        logo = credType.logo;
+      }
+    } else {
+      if (logEntry.serverName != null) {
+        title = logEntry.serverName!.name.translate(lang);
+        if (logEntry.serverName!.logo != null) {
+          logo = logEntry.serverName!.logo;
         }
+      }
```

Comment:
In the current design, long server names tend to fall off the screen. This might be something to consider.
![image](https://user-images.githubusercontent.com/17035191/215725808-04687cb7-6e8d-4965-943e-56b81fc06b02.png)